### ISSUE TITLE
`list` operations design

### ISSUE THREAD

**Issue by JunyiYi** [Azure/autorest.cli#12]
I noticed we have a [`list`](https://github.com/Azure/magic-module-specs/blob/master/batchapplication/api.yaml#L181) operation in batchapplication. I would like to name it as `list_by_parent` based on the following assumptions:

1. A top level (or maybe sub-resource as well) resource could be listed by resource group, we have `list_by_resource_group` support.
2. A sub-resource could be listed by its parent resource (e.g. batchaccount is batchapplication's parent), I would like to introduce `list_by_parent` to `api.yaml` for each sub-resource.

Correct me if you know there are other cases for this scenario.

If ALL sub-resources are not able to be listed by resource group, actually we might also combine those two into one `list_by_parent`, because we can treat `resource_group` as the parent of all top-level resources.

**Comment by zikalino** [comment]
could we just call it **list**? that would be consistent with swagger specs.
actually I can rename it to anything :-)

**Comment by zikalino** [comment]
linking issue: https://github.com/Azure/magic-module-specs/issues/7

**Comment by JunyiYi** [comment]
I prefer `list_by_parent`. And in the [cosmos db case](https://github.com/Azure/magic-module-specs/blob/master/cosmosdbdatabaseaccount/api.yaml#L280), this function should be named as `list_by_parent` in the `cosmos db sql database` resource.

**Comment by zikalino** [comment]
in case of top level resources, should we also have **list_by_parent** or **list_by_subscription**?

**Comment by JunyiYi** [comment]
Yes we should for list_by_subscription. 👍 Top level resource does not have parent resource, so we do not need that.

**Comment by zikalino** [comment]
also should we assume that any **list_by_[something else]** functions be autogenerated too?

**Comment by JunyiYi** [comment]
Nope, MM will not consider those. MM will only recognize the three `list_` functions mentioned above.

**Comment by zikalino** [comment]
will they be ignored? or cause an error?

**Comment by JunyiYi** [comment]
They will cause errors. So please remove them.
